src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -115,16 +121,11 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request, op
 // pipeData pipes data between the client and proxy connections.  It's also
 // responsible for responding to the initial CONNECT request with a 200 OK.
 func pipeData(clientConn net.Conn, connOut net.Conn, op *ops.Op, closeFunc func()) {
-	writeErrCh := make(chan error, 1)
-	// Start piping from client to proxy
-	op.Go(func() {
-		_, writeErr := io.Copy(connOut, clientConn)
-		writeErrCh <- writeErr
-	})
-
-	// Then start copying from proxy to client.
-	_, readErr := io.Copy(clientConn, connOut)
-	writeErr := <-writeErrCh
```

Comment:
Wait sorry but what's bad about not closing the connection until both reading and writing are done? Why would we want to close them before they're done or before the idle timeout @oxtoacart?